common/perception_utils/include/perception_utils/matching.hpp

```diff
@@ -0,0 +1,109 @@
+// Copyright 2022 TIER IV, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef PERCEPTION_UTILS__MATCHING_HPP_
+#define PERCEPTION_UTILS__MATCHING_HPP_
+
+#include "perception_utils/geometry.hpp"
+#include "tier4_autoware_utils/geometry/boost_geometry.hpp"
+#include "tier4_autoware_utils/geometry/boost_polygon_utils.hpp"
+
+#include <boost/geometry.hpp>
+
+#include <vector>
+
+namespace perception_utils
+{
+template <class T1, class T2>
+inline double get2dIoU(const T1 source_object, const T2 target_object)
```

Comment:
I think `inline` is redundant for templates (except for specialization).

```suggestion
double get2dIoU(const T1 source_object, const T2 target_object)
```

Reply:
fixed at [ca81de8](https://github.com/autowarefoundation/autoware.universe/pull/1954/commits/ca81de8a897e2e3c1a398bfc8edd03a77804d4cd)